python/sglang/srt/entrypoints/openai/serving_chat.py

```diff
@@ -542,26 +543,36 @@ async def _generate_chat_stream(
                         )
                         yield f"data: {chunk.model_dump_json()}\n\n"
 
-            # Final chunk with finish_reason
-            finish_reason_chunk = ChatCompletionStreamResponse(
-                id=content["meta_info"]["id"],
-                created=int(time.time()),
-                choices=[
-                    ChatCompletionResponseStreamChoice(
-                        index=index,
-                        delta=DeltaMessage(),
-                        finish_reason=finish_reason_type,
-                        matched_stop=(
-                            finish_reason["matched"]
-                            if finish_reason and "matched" in finish_reason
-                            else None
-                        ),
-                    )
-                ],
-                model=request.model,
-                usage=None,
-            )
-            yield f"data: {finish_reason_chunk.model_dump_json()}\n\n"
+            # Send finish_reason chunks for each index that completed
+            for idx, finish_reason_data in finish_reasons.items():
+                finish_reason_type = finish_reason_data["type"]
+
+                # Change finish_reason to "tool_calls" if we had tool calls and stopped naturally
+                final_finish_reason = finish_reason_type
+                if has_tool_calls.get(idx, False) and finish_reason_type == "stop":
```

Comment:
Not fully sure about this.

\Shall we do this instead? For example, if finish_reason = "length", should we transform the finish_reason to tool_calls too?
```suggestion
                if has_tool_calls.get(idx, False) and finish_reason_type is not None:
```

Reply:
no. If finish_reason is `length`, that means the tool call is not completed. OpenAI doesn't change it to tool_calls in this case.

Reply:
got it. thanks!